### ISSUE TITLE
Some AIO URIs return a Firebase 404 instead of an Angular one

### ISSUE THREAD

**Issue by StephenFluin** [angular/angular#23179]
## I'm submitting a...

<pre><code>
[X] Bug report  
</code></pre>

## Current behavior
Visit https://angular.io/guide/docs-style-guide/docs-style-guide-Stackblitz.png

See a Firebase 404 error.

![image](https://user-images.githubusercontent.com/165056/38348239-819fd53c-3855-11e8-972f-a5f3027598a1.png)

## Expected behavior
See an Angular 404 error

**Comment by gkalpak** [comment]
This happens because we only rewrite URLs to `/index.html` if they don't contains an file extension.
I wouldn't rewrite other URLs to `/index.html`, because that would make them look like 200 responses, but we could add an aio-themed `404.html` page.

**Comment by gkalpak** [comment]
Hm...although it would not be straightforward to keep this consistent with `/index.html` (in looks and behavior) 😕

**Comment by petebacondarwin** [comment]
> because that would make them look like 200 responses, 

I don't think this is true, since we create "soft-404s" for non-existent pages.

So redirecting all 404 pages to `index.html` would seem a reasonable approach and would keep the project simpler, rather than having to generate a static 404 page at build time.

**Comment by gkalpak** [comment]
I don't think it is a good idea to have soft 404s for resources. Soft 404s are still technically 200s and resources could be requested not as top-level documents (i.e. navigation requests) but as....resources (e.g. images).

**Comment by petebacondarwin** [comment]
Images in the docs are checked by dgeni so we shouldn't have any 404s from doc pages.

**Comment by gkalpak** [comment]
Not sure what the usecase is, but I am pretty sure it is not about us referencing nob-existent resources ('cause we would never do that :grin:)
